types/vec2/index.d.ts

```diff
@@ -0,0 +1,191 @@
+
+    /**
+     * Normalizes this vector.  If `returnNew` is truthy, a new vector populated with the normalized coordinates will be returned.
+     */
+    public normalize(returnNew?: boolean): Vec2;
+
+    /**
+     * Returns true if the incoming coordinates are the same as this vector's
+     */
+    public equal(x: number, y: number): boolean;
+    public equal(arrayVec2: number[] | Vec2): boolean;
+
+    /**
+     * Return a `Vec2` that contains the absolute value of each of this vector's parts.
+     * If `returnNew` is truthy, create a new `Vec2` and return it. Otherwise apply the absolute values to to `this`.
+     */
+    public abs(returnNew?: boolean): Vec2;
+
+    /**
+     * Return a `Vec2` consisting of the smallest values from this vector and the incoming
+     * When returnNew is truthy, a new `Vec2` will be returned otherwise the minimum values in either this or `vec` will be applied to this vector.
+     */
+    public min(vec: Vec2, returnNew?: boolean): Vec2;
+
+    /**
+     * Return a `Vec2` consisting of the largest values from this vector and the incoming
+     * When returnNew is truthy, a new `Vec2` will be returned otherwise the maximum values in either `this` or `vec` will be applied to this vector.
+     */
+    public max(vec: Vec2, returnNew?: boolean): Vec2;
+
+    /**
+     * Clamp the coordinates of this vector to the high/low of the incoming vec2s.  If `returnNew` apply the result to the new vector and return.
+     * Otherwise apply to this vector.
+     */
+    public clamp(low: Vec2, high: Vec2, returnNew?: boolean): Vec2;
+
+    /**
+     *
+     * Perform linear interpolation between this vector and the incoming.
+     * `amount` - the percentage along the path to place the vector
+     * `returnNew` - if `truthy`, apply the result to a new vector and return it, otherwise return `this`
+     */
+    public lerp(vec: Vec2, amount: number, returnNew?: boolean): Vec2;
+
+    /**
+     * Returns a vector set with the `(-y,x)` coordinates of this vector.  If `returnNew` a new vector is created and the operation is applied to the new vector.
+     */
+    public skew(returnNew?: boolean): Vec2;
+
+    /**
+     * _returns_: `double`
+     */
+    public dot(): number;
+
+    /**
+     * _returns_: `double`
+     */
+    public perpDot(): number;
+
+    /**
+     * Returns the angle from this vector to the incoming.
+     */
+    public angleTo(vec: Vec2): number;
+
+    /**
+     * Where `start` and `end` are vec2-like (e.g. `start.x` and `start.y`)
+     */
+    public isPointOnLine(start: Vec2, end: Vec2): boolean;
+
+    /**
+     * _returns_: `[x, y]`
+     */
+    public toArray(): number[];
+
+    /**
+     * Applies the `[0]` to `this.x` and `[1]` to `this.y`
+     */
+    public fromArray(array: number[]): Vec2;
+
+    public toJSON(): { x: number, y: number };
+
+    /**
+     * _returns_: `'(x, y)'`
+     */
+    public toString(): string;
+}
+
+export default Vec2;
```

Comment:
Looking at 
https://unpkg.com/vec2@1.6.0/vec2.js, it looks like the shipped version has no default property on module.exports.  So it should be `export = Vec2`. 

Probably true of polygon too.

Reply:
Ok. I changed it like you have proposed.

But for polygon i have to write `export default Polygon`.
If i use `export = Polygon`, i get the error `Module '.../DefinitelyTyped/types/polygon/index"' has no default export.` running `npm test`

Reply:
well, I think the emitted JS code will be `var Vec2 = require("vec2")`, so the equivalent Typescript import should be `import Vec2 = require("vec2")` not `import Vec2 from "vec2"`.

Reply:
Yeah, this is exactly what i tried and got the mentioned errors. See this [TravisRun](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/507749764?utm_source=github_status&utm_medium=notification)
I applied your proposal for vec2 and polygon. For vec2 only, the error is also present.

(Sorry for my late response, was very sick last week)

Reply:
I don't think the PR ever had all four of the import/exports in the following state, so the build was always reporting mismatches:

```
export = Vec2
import vec2 = require("vec2")

export = Polygon
import polygon = require("polygon")
```

(I guess it is five actually, since polygon depends on vec2 and has an import statement for it.)

If they all match, that should resolve the errors.

Reply:
Ouh. My bad. Forgot the correct import in the test files 🙈 
Now everything is working!